Hast.Communication/MemberInvocationHandlerFactory.cs

```diff
@@ -116,9 +116,15 @@ async Task InvocationHandler()
                         var memory = (SimpleMemory)invocation.Arguments.SingleOrDefault(argument => argument is SimpleMemory);
                         if (memory != null)
                         {
-                            foreach (var checker in memoryResourceCheckers)
+                            if (memoryResourceCheckers.Check(memory, hardwareRepresentation) is { } problem)
                             {
-                                checker.EnsureResourceAvailable(memory, hardwareRepresentation);
+                                var exception = new InvalidOperationException(
+                                    $"The input is too large to fit into the device's memory. The input is " +
+                                    $"{problem.MemoryByteCount} bytes, the available memory is " +
+                                    $"{problem.AvailableByteCount} bytes. (reported by {problem.GetType().FullName}) " +

```

Comment:
Ez a ponttal lezárt mondat után kis betűvel kezdünk a zárójelen belül egy visszatérő becsípődés :D.

Reply:
Hát a zárójelen belüli rész nem egy teljes mondat ezért furcsa nagybetűvel kezdeni, viszont egy külön gondolat aminek semmi köze az előző mondathoz ezért nem lenne helyes bevinni a pont elé. se Akkor már inkább
```csharp
$"{problem.AvailableByteCount} bytes. (Reported by {problem.GetType().FullName}.) "
```

Reply:
Hát csonka mondat is mondat, szóval attól még naggyal írjuk és írásjelet teszünk a végére :). De így is OK.